x/inflation/keeper/hooks.go

```diff
@@ -80,6 +82,16 @@ func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumb
 		k.SetEpochMintProvision(ctx, newProvision)
 	}
 
+	defer func() {
```

Comment:
The cognitive complexity score here becomes 8 with this addition which is just above 7 as set by the lint.
It may be okay now but may require a refactor when the lint is activated. Unless the limits are moved from 7.